homeassistant/components/enocean/cover.py

```diff
@@ -0,0 +1,267 @@
+    ATTR_POSITION,
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASSES_SCHEMA,
+    PLATFORM_SCHEMA,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.const import CONF_DEVICE_CLASS, CONF_ID, CONF_NAME
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import SIGNAL_SEND_MESSAGE
+from .device import EnOceanEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "EnOcean roller shutter"
+DEPENDENCIES = ["enocean"]
+
+CONF_SENDER_ID = "sender_id"
+WATCHDOG_TIMEOUT = "watchdog_timeout"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Required(CONF_SENDER_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(WATCHDOG_TIMEOUT, default=5): int,
+    }
+)
+
+
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Cover platform for EnOcean."""
+    dev_id = config.get(CONF_ID)
+    sender_id = config.get(CONF_SENDER_ID)
+    dev_name = config.get(CONF_NAME)
+    device_class = config.get(CONF_DEVICE_CLASS)
+    if device_class is None:
+        device_class = DEVICE_CLASS_BLIND
+    watchdog_timeout = config.get(WATCHDOG_TIMEOUT)
+    add_entities(
+        [
+            EnOceanCover(
+                sender_id, dev_id, dev_name, device_class, hass, watchdog_timeout
+            )
+        ]
+    )
+
+
+class EnOceanCover(EnOceanEntity, CoverEntity):
+    """Representation of an EnOcean Cover (EEP D2-05-00)."""
+
+    def __init__(
+        self, sender_id, dev_id, dev_name, device_class, hass, watchdog_timeout
+    ):
+        """Initialize the EnOcean Cover."""
+        super().__init__(dev_id, dev_name)
+        self._device_class = device_class
+        self._position = None
+        self._is_closed = None
+        self._is_opening = False
+        self._is_closing = False
+        self._sender_id = sender_id
+        self._dev_name = dev_name
+        self._attr_unique_id = f"{combine_hex(dev_id)}-{device_class}"
+        self._state_changed_by_command = False
+        self._watchdog_enabled = False
+        self._watchdog_seconds_remaining = 0
+        self._watchdog_timeout = watchdog_timeout
+        self._hass = hass
+
+    @property
+    def name(self):
+        """Return the default name for the cover."""
+        return self._dev_name
+
+    @property
+    def device_class(self):
+        """Return the class of this cover."""
+        return self._device_class
```

Comment:
We prefer using our shorthand attributes nowadays when possible.

https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes

Reply:
Ok, see commit [9b62de4](https://github.com/home-assistant/core/pull/75356/commits/9b62de4af69bc738dcfbc47ba66c92a036e11dd7)